src/main/java/net/logstash/logback/composite/loggingevent/SequenceProvider.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.composite.loggingevent;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import com.fasterxml.jackson.core.JsonGenerator;
+import net.logstash.logback.composite.AbstractFieldJsonProvider;
+import net.logstash.logback.composite.JsonWritingUtils;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Outputs an incrementing sequence number.
+ * Useful for determining if log events get lost along the transport chain.
+ */
+public class SequenceProvider extends AbstractFieldJsonProvider<ILoggingEvent> {
+
+    public static final String FIELD_SEQUENCE = "sequence";
+
+    private static AtomicLong sequenceNumber = new AtomicLong(0L);
```

Comment:
This should be `final`.

Also, did you intend to make this static, so that the sequence is global to the entire application?  It seems like this should be an instance field, so the `sequenceNumber` is per provider instance.  This would allow you to configure two different encoders (possibly to two different destinations), each with their own sequences, to keep track of them separately.